pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -228,6 +327,23 @@ func generateYAMLWithSettings(cfg echo.Config, settings *image.Settings) (servic
 		}
 	}
 
+	var istiodIP, istiodPort string
+	if cfg.DeployAsVM {
+		s, err := kube.NewSettingsFromCommandLine()
+		if err != nil {
+			return "", "", err
+		}
+		addr, err := istio.GetRemoteDiscoveryAddress("istio-system", cluster, s.Minikube)
```

Comment:
nit: for the future you may wish to get rid of the implementation specific naming in the test cases `Minikube`.